src/main/java/com/openlattice/chronicle/util/ChronicleServerUtil.java

```diff
@@ -45,6 +45,31 @@ public static UUID getFirstUUIDOrNull( Map<FullQualifiedName, Set<Object>> entit
         }
     }
 
+    public static String getLoggingMessage( String message, UUID orgId, UUID studyId, String participantId ) {
```

Comment:
This should probably be replaced by using the Log4j Logger built in string templating. Instead of doing the below using String.format(), you can just do this:
```
" - orgId = {}, studyId = {}, participantId = {}"
```
and include it as part of the string passed to `logger.error()` with orgId, studyId and participantId passed as arguments after the message.

```
logger.error("Error logging data  - orgId = {}, studyId = {}, participantId = {}", organizationId, studyId, participantId)
```

So a better way to create a helper for this use case would probably be like this:
```
public static final string orgStudyParticipantLoggingString = " - orgId = {}, studyId = {}, participantId = {}";
public static final string orgStudyParticipantDatasourceLoggingString = " - orgId = {}, studyId = {}, participantId = {}, datasourceId = {}";

public static void logVerboseError(
            Logger logger, 
            String message,
            UUID orgId,
            UUID studyId,
            String participantId,
            String datasourceId) {
            logger.error(message + orgStudyParticipantDatasourceLoggingString, orgId, studyId, participantId, datasourceId);
}
```

Reply:
➕

Reply:
This is not always used for `logger.error` so I want to return a string instead of calling `logger.error` in here.

Reply:
Regardless of which log4j call you're making though, this should still use the log4j syntax. This particular snippet of the log seems small enough that it's not a big deal to just include it in each logging statement, but if you're trying to constantify it somehow, it's much cleaner to just create something like `private const ORG_STUDY_PARTICIPANT = " - orgId = {}, studyId = {}, participantId = {}"` and pass those params to the logger -- i.e. `logger.info( "whatever message" + ORG_STUDY_PARTICIPANT, orgId, studyId, participantId)` -- instead of this helper function.

Alternatively you could pass an indicator of the log level or the logging function itself, although my suggestions in the paragraph above would be preferable.